examples/distributed/homogeneous_training.py

```diff
@@ -0,0 +1,898 @@
+            subgraph_fanout,  # subgraph_fanout
+            sampling_workers_per_process,  # sampling_workers_per_process
+            main_batch_size,  # main_batch_size
+            random_batch_size,  # random_batch_size
+            sampling_worker_shared_channel_size,  # sampling_worker_shared_channel_size
+            process_start_gap_seconds,  # process_start_gap_seconds
+            use_automatic_mixed_precision,  # use_automatic_mixed_precision
+            log_every_n_batch,  # log_every_n_batch
+            learning_rate,  # learning_rate
+            weight_decay,  # weight_decay
+            num_max_train_batches,  # num_max_train_batches
+            num_val_batches,  # num_val_batches
+            val_every_n_batch,  # val_every_n_batch
+        ),
+        nprocs=local_world_size,
+        join=True,
+    )
+    logger.info(f"--- Training finished, took {time.time() - start_time} seconds")
+    logger.info("--- Launching test processes ...\n")
+    start_time = time.time()
+    torch.multiprocessing.spawn(
+        _testing_process,
+        args=(  # Corresponding arguments in `_testing_process` function
+            local_world_size,  # local_world_size
+            node_rank,  # node_rank
+            node_world_size,  # node_world_size
+            dataset,  # dataset
+            node_feature_dim,  # node_feature_dim
+            edge_feature_dim,  # edge_feature_dim
+            master_ip_address,  # master_ip_address
+            master_process_group_port_for_testing,  # master_default_process_group_port
+            trainer_args,  # trainer_args
+            model_uri,  # model_uri
+            subgraph_fanout,  # subgraph_fanout
+            sampling_workers_per_process,  # sampling_workers_per_process
+            main_batch_size,  # main_batch_size
+            random_batch_size,  # random_batch_size
+            sampling_worker_shared_channel_size,  # sampling_worker_shared_channel_size
+            process_start_gap_seconds,  # process_start_gap_seconds
+            use_automatic_mixed_precision,  # use_automatic_mixed_precision
+            log_every_n_batch,  # log_every_n_batch
+        ),
+        nprocs=local_world_size,
+        join=True,
+    )
+    logger.info(f"--- Testing finished, took {time.time() - start_time} seconds")
+    logger.info("--- All steps finished, exiting main ---")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Arguments for distributed model training on VertexAI"
+    )
+    parser.add_argument("--task_config_uri", type=str, help="Gbml config uri")
+
+    # We use parse_known_args instead of parse_args since we only need job_name and task_config_uri for distributed trainer
+    args, unused_args = parser.parse_known_args()
+    logger.info(f"Unused arguments: {unused_args}")
+
+    # We only need `task_config_uri` for running trainer
+    _run_example_training(
+        task_config_uri=args.task_config_uri,
+    )
```

Comment:
```suggestion
    )

```

Does our linter not pick up these files? I can look into that...